### PR TITLE
Convert to use new Shared AppVeyor and Tests - Fixes #78, Fixes #79

### LINKED ISSUE #78: Convert to use new Shared Deploy/Test Code in DSCResource.tests
[This PR](https://github.com/PowerShell/DscResource.Tests/pull/90) added shared AppVeyor deployment and Wiki generation code into DSCResource.Tests.

Update this repo to use this new code.

### PR DESCRIPTION
**Pull request by PlagueHO**
This PR:
- Converted AppVeyor build process to use AppVeyor.psm1.
- Added support for auto generating wiki, help files, markdown linting
  and checking examples.
- Correct name of MSFT_xDiskAccessPath.tests.ps1.
- Move shared modules into Modules folder.
- Fixed unit tests.
- Removed support for WMI cmdlets.
- Opted in to Markdown and Example tests.

Fixes #78 
Fixes #79

---
This change is [Reviewable](https://reviewable.io/reviews/powershell/xstorage/79)

### CODE REVIEW

**Comment by johlju** [comment]
@PlagueHO Any fixes you do over at xNetworking, could you port them here too? I can help you review this one when that is done.

**Comment by PlagueHO** [comment]
@johlju - I've ported all my xNetworking changes over here as well now - so should be good to go.

File: Modules/xStorage/DSCResources/MSFT_xDiskAccessPath/MSFT_xDiskAccessPath.psm1 [hunk 1]
```diff
@@ -524,7 +511,7 @@ function Test-TargetResource
     {
         Write-Verbose -Message ( @(
                 "$($MyInvocation.MyCommand): "
-                $($LocalizedData.DiskNotGPTMessage -f $DiskNumber,$Disk.PartitionStyle)
+                $($localizedData.DiskNotGPTMessage -f $DiskNumber,$Disk.PartitionStyle)
```
Comment (TravisEz13) [inline Modules/xStorage/DSCResources/MSFT_xDiskAccessPath/MSFT_xDiskAccessPath.psm1]: It would have been good to submit this as a separate PR.

File: Tests/TestHarness.psm1 [hunk 2]
```diff
@@ -0,0 +1,54 @@
+function Invoke-TestHarness
```
Comment (TravisEz13) [inline Tests/TestHarness.psm1]: Why do we need this it seems equivalent to the default.

File: README.md [hunk 3]
```diff
@@ -1,364 +1,23 @@
...
-
-Please leave comments, feature requests, and bug reports in the Q & A tab for this module.
-
-If you would like to modify this module, feel free. When modifying, please update the module name, resource friendly name, and MOF class name (instructions below). As specified in the license, you may copy or modify this resource as long as they are used on the Windows Platform.
-
-For more information about Windows PowerShell Desired State Configuration, check out the blog posts on the [PowerShell Blog](http://blogs.msdn.com/b/powershell/) ([this](http://blogs.msdn.com/b/powershell/archive/2013/11/01/configuration-in-a-devops-world-windows-powershell-desired-state-configuration.aspx) is a good starting point). There are also great community resources, such as [PowerShell.org](http://powershell.org/wp/tag/dsc/), or [PowerShell Magazine](http://www.powershellmagazine.com/tag/dsc/). For more information on the DSC Resource Kit, checkout [this blog post](http://go.microsoft.com/fwlink/?LinkID=389546).
-
-## Installation
```
Comment (TravisEz13) [inline README.md]: Minor, I think it might be good to have a link to at least common installation instructions.

File: Modules/xStorage/DSCResources/MSFT_xDiskAccessPath/MSFT_xDiskAccessPath.psm1 (hunk 1 above)
Comment (PlagueHO) [inline Modules/xStorage/DSCResources/MSFT_xDiskAccessPath/MSFT_xDiskAccessPath.psm1]: @TravisEz13 - sorry about that - I was actually working on the HQRM for this but once the DSCResource.Tests went through it broke my HQRM PR so I ended up combining them. Again, sorry about that.

File: Tests/TestHarness.psm1 (hunk 2 above)
Comment (PlagueHO) [inline Tests/TestHarness.psm1]: It does have some behavior differences required because the resources and examples are all moved into a subfolder within the repo rather than being in the root. I've just copied the harness code almost straight over from SharePointDsc and xNetworking.

File: README.md (hunk 3 above)
Comment (PlagueHO) [inline README.md]: I only removed this because it seems like all the other repos had removed it (e.g. https://github.com/PowerShell/xWindowsUpdate). Happy to add it back in, but we should have some sort of standard instruction we include in all repos (for consistency). Your call! :grin:

Comment (TravisEz13) [inline README.md]: @kwirkykat what do you think?  I was thinking a section on installation in dscresources and a link here to that.

Comment (PlagueHO) [inline README.md]: All good with doing what ever @kwirkykat suggests. What would be good would be to add a standard text snippet to DSCResources so we could copy/paste it into all repos.

Comment (TravisEz13) [inline README.md]: Don't consider this blocking.  I've seen this go in this way on other repos

Comment (PlagueHO) [inline README.md]: Cool. Thanks @TravisEz13.

Will just wait on a review from @johlju (but he's been so busy reviewing all my other PR's of late I'd forgive this waiting for a while :grin: )

Comment (kwirkykat) [inline README.md]: @TravisEz13 @PlagueHO This is fine to remove. We now have common installation instructions in the [DscResources main README](https://github.com/powershell/DSCResources#released-dsc-resource-modules)

Comment (PlagueHO) [inline README.md]: Cool. Thanks @kwirkykat

**Comment by PlagueHO** [comment]
Hi @TravisEz13 , @kwirkykat - this should be good to go on completing the review if either of you get a chance.

**Comment by PlagueHO** [comment]
Thanks @TravisEz13 !

**Comment by johlju** [comment]
@PlagueHO Sorry for not having the time to help you review this one. Extremely busy last two weeks and all free time went to fixing problems and doing review over at xSQLServer :)
Glad you got this thru with @TravisEz13's help!

**Comment by PlagueHO** [comment]
@johlju - no worries! I know how busy xSQLServer is - I see the number of PRs you get through - extremely impressive - no idea how you manage to actually sleep. :grin:

**Comment by johlju** [comment]
@PlagueHO Thanks! But actually it is not me that are impressive. It is all the contributors that constantly pushing new code, so they are keeping me busy... that is impressive, and fun! 😄
